samples_project/Assets/SampleViewer/Samples/VRSample/Scripts/ContinuousMovement.cs

```diff
@@ -28,25 +62,74 @@ void Start()
     private void FixedUpdate()
     {
         FollowHeadset();
-        Quaternion headYaw = Quaternion.Euler(0, rig.Camera.transform.eulerAngles.y, 0);
-        Vector3 direction = headYaw * new Vector3(leftInputAxis.x, 0, leftInputAxis.y);
-        Vector3 up = new Vector3(0, rightInputAxis.y, 0);
-        controller.Move(direction * speed * Time.fixedDeltaTime);
-        controller.Move(up * upSpeed * Time.fixedDeltaTime);
+        var headYaw = Quaternion.Euler(0, rig.Camera.transform.eulerAngles.y, 0);
+        var direction = headYaw * new Vector3(leftInputAxis.x, 0, leftInputAxis.y);
+        var up = new Vector3(0, rightInputAxis.y, 0);
+        controller.Move(speed * Time.fixedDeltaTime * direction);
+        controller.Move(upSpeed * Time.fixedDeltaTime * up);
+    }
 
+    private void LateUpdate()
+    {
+        UICanvas.transform.position = playerCamera.transform.position + playerCamera.transform.forward * 15;
+        UICanvas.transform.rotation = new Quaternion(0.0f, playerCamera.transform.rotation.y, 0.0f, playerCamera.transform.rotation.w);
     }
     // Update is called once per frame
     void Update()
     {
-        InputDevice leftDevice = InputDevices.GetDeviceAtXRNode(leftInputSource);
+        var leftDevice = InputDevices.GetDeviceAtXRNode(leftInputSource);
```

Comment:
You should cache `InputDevices.GetDeviceAtXRNode(leftInputSource)` and `InputDevices.GetDeviceAtXRNode(rightInputSource)` in member variables at the top of the script

Reply:
I would still do this one ☝️